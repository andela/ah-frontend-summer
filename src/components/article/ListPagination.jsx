import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Pagination, Grid } from 'semantic-ui-react';
import '../../styles/Articles.scss';


class ListPagination extends Component {
    static currentLocalStorage = (localStorage) => {
        const initialPage = 1;
        let currentPage = localStorage.getItem('page');
        if (currentPage !== null) {
            return currentPage;
        } else {
            return currentPage = initialPage;
        }
    }
    maxPageCount = 10;
    onPageChange = (event, data) => {
        const { fetchArticles } = this.props;
        const url = 'https://ah-backend-summer-staging.herokuapp.com/api/v1/articles';
        const initialIndex = 0;
        const limit = (count, index) => 
            `limit=${count}&offset=${index ? index * count : initialIndex}`;
        const page = data.activePage;
        const startPage = 1;
        const pagination = `?${limit(this.maxPageCount, page - startPage)}`;
        fetchArticles(url + pagination);
        localStorage.setItem('page', page);
    };
    render() {
        const { articleCount } = this.props;
        const pageNumber = Math.ceil(articleCount / this.maxPageCount);
        const currentPage = ListPagination.currentLocalStorage(localStorage);
        localStorage.removeItem('page');
        return (
            <Grid>
                <Grid.Row columns={3}>
                    <Grid.Column width={3}></Grid.Column>
                    <Grid.Column width={10} textAlign="center" st>
                        <Pagination
                            boundaryRange={0}
                            activePage={currentPage}
                            ellipsisItem={null}
                            firstItem={null}
                            lastItem={null}
                            siblingRange={1}
                            totalPages={pageNumber}
                            onPageChange={this.onPageChange}
                        />
                    </Grid.Column>
                    <Grid.Column width={3}></Grid.Column>
                </Grid.Row>
            </Grid>
        );
    }
}

ListPagination.propTypes = {
    fetchArticles: PropTypes.func.isRequired,
    articleCount: PropTypes.number.isRequired
};

export default ListPagination;
