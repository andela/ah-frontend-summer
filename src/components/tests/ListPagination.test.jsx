import React from 'react';
import { shallow } from 'enzyme';
import ListPagination from '../article/ListPagination';

let onPageChange, fetchArticles, articleCount, wrapper;

beforeEach(() => {
    onPageChange = jest.fn();
    fetchArticles = jest.fn();
    articleCount = 20;
    wrapper = shallow(
        <ListPagination
            onPageChange={onPageChange}
            fetchArticles={fetchArticles}
            articleCount={articleCount}
        />
    );
});

describe('ListPagination tests', () => {
    it('should render ListPagination correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle page change', () => {
        const event = null;
        const data = {};
        wrapper.find('Pagination').prop('onPageChange')(event, data);
        expect(fetchArticles).toHaveBeenCalled();
    });

    it('should set local storage', () => {
        const localStorage = {};
        localStorage.getItem = jest.fn().mockImplementation(() => { return 2;});
        expect(ListPagination.currentLocalStorage(localStorage)).toBe(2);
        expect(localStorage.getItem).toHaveBeenCalled();
        expect(localStorage.getItem.mock.calls.length).toBe(1);
    });
});
