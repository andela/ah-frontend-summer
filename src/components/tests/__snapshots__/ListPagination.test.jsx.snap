// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPagination tests should render ListPagination correctly 1`] = `
<Grid>
  <GridRow
    columns={3}
  >
    <GridColumn
      width={3}
    />
    <GridColumn
      st={true}
      textAlign="center"
      width={10}
    >
      <Pagination
        activePage={1}
        aria-label="Pagination Navigation"
        boundaryRange={0}
        ellipsisItem={null}
        firstItem={null}
        lastItem={null}
        nextItem={
          Object {
            "aria-label": "Next item",
            "content": "⟩",
          }
        }
        onPageChange={[Function]}
        pageItem={Object {}}
        prevItem={
          Object {
            "aria-label": "Previous item",
            "content": "⟨",
          }
        }
        siblingRange={1}
        totalPages={2}
      />
    </GridColumn>
    <GridColumn
      width={3}
    />
  </GridRow>
</Grid>
`;
